bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/editor/expressions/ClangdLanguageServerProviderActive.java

```diff
@@ -0,0 +1,20 @@
+package org.eclipse.cdt.lsp.internal.clangd.editor.expressions;
+
+import org.eclipse.cdt.lsp.LspPlugin;
+import org.eclipse.cdt.lsp.internal.clangd.ClangdLanguageServerProvider;
+import org.eclipse.core.expressions.PropertyTester;
+
+public class ClangdLanguageServerProviderActive extends PropertyTester {
+	private final boolean clangdLanguageServerProviderActive;
```

Comment:
I'm not sure that we need to cache result in a field. This check should be too heavy and I also I expect further customization work near `LspPlugin.getDefault().getCLanguageServerProvider()`